pytests/tests/test_enums.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+from pyo3_pytests import enums
+
+
+def test_complex_enum_variant_constructors():
+    int_variant = enums.ComplexEnum.Int(42)
+    assert isinstance(int_variant, enums.ComplexEnum.Int)
+
+    float_variant = enums.ComplexEnum.Float(3.14)
+    assert isinstance(float_variant, enums.ComplexEnum.Float)
+
+    str_variant = enums.ComplexEnum.Str("hello")
+    assert isinstance(str_variant, enums.ComplexEnum.Str)
+
+    empty_struct_variant = enums.ComplexEnum.EmptyStruct()
+    assert isinstance(empty_struct_variant, enums.ComplexEnum.EmptyStruct)
+
+    multi_field_struct_variant = enums.ComplexEnum.MultiFieldStruct(42, 3.14, True)
+    assert isinstance(multi_field_struct_variant, enums.ComplexEnum.MultiFieldStruct)
+
+
+@pytest.mark.parametrize(
+    "variant",
+    [
+        enums.ComplexEnum.Int(42),
+        enums.ComplexEnum.Float(3.14),
+        enums.ComplexEnum.Str("hello"),
+        enums.ComplexEnum.EmptyStruct(),
+        enums.ComplexEnum.MultiFieldStruct(42, 3.14, True),
+    ],
+)
+def test_complex_enum_variant_subclasses(variant: enums.ComplexEnum):
+    assert isinstance(variant, enums.ComplexEnum)
+
+
+def test_complex_enum_field_getters():
+    int_variant = enums.ComplexEnum.Int(42)
+    assert int_variant.i == 42
+
+    float_variant = enums.ComplexEnum.Float(3.14)
+    assert float_variant.f == 3.14
+
+    str_variant = enums.ComplexEnum.Str("hello")
+    assert str_variant.s == "hello"
+
+    multi_field_struct_variant = enums.ComplexEnum.MultiFieldStruct(42, 3.14, True)
+    assert multi_field_struct_variant.a == 42
+    assert multi_field_struct_variant.b == 3.14
+    assert multi_field_struct_variant.c is True
+
+
+@pytest.mark.parametrize(
+    "variant",
+    [
+        enums.ComplexEnum.Int(42),
+        enums.ComplexEnum.Float(3.14),
+        enums.ComplexEnum.Str("hello"),
+        enums.ComplexEnum.EmptyStruct(),
+        enums.ComplexEnum.MultiFieldStruct(42, 3.14, True),
+    ],
+)
+def test_complex_enum_match_statement(variant: enums.ComplexEnum):
+    match variant:
+        case enums.ComplexEnum.Int(i=x):
+            assert x == 42
+        case enums.ComplexEnum.Float(f=x):
+            assert x == 3.14
+        case enums.ComplexEnum.Str(s=x):
+            assert x == "hello"
+        case enums.ComplexEnum.EmptyStruct():
+            assert True
+        case enums.ComplexEnum.MultiFieldStruct(a=x, b=y, c=z):
+            assert x == 42
+            assert y == 3.14
+            assert z is True
+        case _:
+            assert False
```

Comment:
@davidhewitt I think the merge queue checks fail because this `match` syntax is only available in Python 3.10 and up. What would be a good way to fix this?

Reply:
Agreed, yes. I am not sure what the right solution should be. Are you aware of any way to make `pytest` ignore files based on a version pattern?

Reply:
@davidhewitt I think I've managed to fix the problem. Just waiting for CI to finish... 🤞